### PR TITLE
build02: add MCP pthread-debug

### PR DESCRIPTION
**Pull request by hongkailiu**
Ref.

https://github.com/openshift/machine-config-operator/blob/master/docs/ContainerRuntimeConfigDesign.md
https://github.com/openshift/machine-config-operator/blob/master/docs/custom-pools.md

/cc @alvaroaleman @mrunalp 

/hold

Need to label the node before merge:

```
oc --as system:admin --context build02 label node pthread-debug-k8tj2.c.openshift-ci-build-farm.internal node-role.kubernetes.io/pthread-debug=
```

I will check on the node (after merge) if the CRC takes effect.

This is the current value:

```
oc --as system:admin --context build02 debug node/pthread-debug-k8tj2.c.openshift-ci-build-farm.internal
Starting pod/pthread-debug-k8tj2copenshift-ci-build-farminternal-debug ...
To use host binaries, run `chroot /host`
Pod IP: 10.0.32.24
If you don't see a command prompt, try pressing enter.
sh-4.2# chroot /host
sh-4.4# cat /etc/crio/crio.conf | grep pids_limit
pids_limit = 1024
```

### CODE REVIEW

File: clusters/build-clusters/02_cluster/openshift-machine-api/pthread-debug_crc.yaml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+apiVersion: machineconfiguration.openshift.io/v1
```
Comment (alvaroaleman) [inline clusters/build-clusters/02_cluster/openshift-machine-api/pthread-debug_crc.yaml:1]: Why copy this into a new file?

File: clusters/build-clusters/02_cluster/openshift-machine-api/pthread-debug_mcp.yaml [hunk 2]
```diff
@@ -0,0 +1,13 @@
+apiVersion: machineconfiguration.openshift.io/v1
+kind: MachineConfigPool
```
Comment (alvaroaleman) [inline clusters/build-clusters/02_cluster/openshift-machine-api/pthread-debug_mcp.yaml:2]: Doesn't the MachienSet also need to specify its machineconfigpool?

File: clusters/build-clusters/02_cluster/openshift-machine-api/pthread-debug_crc.yaml (hunk 1 above)
Comment (hongkailiu) [reply clusters/build-clusters/02_cluster/openshift-machine-api/pthread-debug_crc.yaml]: Removed.

File: clusters/build-clusters/02_cluster/openshift-machine-api/pthread-debug_mcp.yaml (hunk 2 above)
Comment (hongkailiu) [reply clusters/build-clusters/02_cluster/openshift-machine-api/pthread-debug_mcp.yaml:2]: https://github.com/openshift/machine-config-operator/blob/master/docs/custom-pools.md
I do not see any machineset mentioned ^^^.

This is my understanding:
MachineSet is a concept of `openshift-machine-api`. It provisions machines/nodes using cloud API.
MachineConfig and MachineConfigPool are from `openshift-machine-config-operator`. They work on nodes level (see the nodeSelector in its spec).

This is from [the MCP doc](https://github.com/openshift/machine-config-operator/blob/master/docs/custom-pools.md):
_Since a custom pool inherits from the worker pool, any change to the worker pool is going to roll out to the custom pool as well (like an OS update during an upgrade)._

**Comment by hongkailiu** [comment]
/hold cancel

```bash
➜  ~ oc --as system:admin --context build02 label node pthread-debug-k8tj2.c.openshift-ci-build-farm.internal node-role.kubernetes.io/pthread-debug=
node/pthread-debug-k8tj2.c.openshift-ci-build-farm.internal labeled


➜  ~ oc --as system:admin --context build02 get node pthread-debug-k8tj2.c.openshift-ci-build-farm.internal
NAME                                                     STATUS   ROLES                  AGE   VERSION
pthread-debug-k8tj2.c.openshift-ci-build-farm.internal   Ready    pthread-debug,worker   8d    v1.18.3+002a51f
```

**Pull request merged**